src/main/kotlin/oop/Proxy/ProtectionProxy.kt

```diff
@@ -1,13 +1,77 @@
 package oop.Proxy
 
+interface Car {
+  fun isNovel(): Boolean
+  fun drive()
+}
+
+class CarProtectionProxy(private val realObject: Car) : Car {
+  override fun drive() {
+    if (!realObject.isNovel()) {
+      realObject.drive()
+    } else {
+      println("You can't drive a car")
+    }
+  }
+
+  override fun isNovel(): Boolean = realObject.isNovel()
+}
+
+class CarVirtualProxy(private val create: () -> Car) : Car {
+
+  private val realObject by lazy {
+    create()
+  }
+
+  override fun drive() = realObject.drive()
+
+  override fun isNovel(): Boolean = realObject.isNovel()
+}
+
+class CarSmartProxy(private val realObject: Car,
+                    var accesDriver: Int = 0) : Car {
+
+  override fun drive() {
+    realObject.drive()
+    accesDriver++
+  }
+
+  override fun isNovel(): Boolean = realObject.isNovel()
+}
+
+class RealCar(val name: String, val driverAge: Int) : Car {
+  override fun drive() = println("$name is driving a car")
+
+  override fun isNovel(): Boolean = driverAge < 18
+}
+
+fun main(args: Array<String>) {
+
+
+  var carVirtualProxy = CarVirtualProxy({ RealCar("tonilopezmr", 5) })
```

Comment:
To be lazy, do you remember? 
Because you don't want to do a `new` inside https://github.com/CloudCoders/Design-Patterns/pull/55/files/ad103bb6a835ebef42ee91260faf98a8889db69e#diff-50695285805be0311b4d9813df63d03dR22